src/geometry.py

```diff
@@ -7,13 +7,19 @@
 
 
 class Geometry(QObject):
-    def __init__(self, guid = None):
+    def __init__(self, guid=None):
         super().__init__()
         self._guid = guid
         if not self._guid:
             self._guid = uuid.uuid4()
 
         self._mesh = om.TriMesh()
+        self.subdivboxtree = 0
+
+    def createSubdivisonBoxTree(self):
+        from subDivBoxTree import SubDivBoxTree
+        self.subdivboxtree = SubDivBoxTree(self._mesh)
+        self.subdivboxtree.createTreeRoot(self.bbox)
```

Comment:
these questions are about the API of `SubDivBoxTree`:

Would you pls elaborate why is this 2 step initialisation needed? What does `createTreeroot` do?

Reply:
The constructor of SubDixBoxTree is called to only create a SubDixBoxTree object. createTreeRoot is called to initialize the division of the geometry mesh into several SubDixBoxTree objects.

Reply:
> The constructor of SubDixBoxTree is called to only create a SubDixBoxTree object. createTreeRoot is called to initialize the division of the geometry mesh into several SubDixBoxTree objects.

Would you please elaborate on this?

It is not clear why is this kind of initialisation needed. Firstly, after constructing it seems that tree is only "half constructed" (antipattern), and secondly, why ordinary copying of trees if not sytable?

Reply:
I am sorry, either I am not understanding what you mean by elaborate or we are misunderstanding each other.

Can you shortly describe what you mean by elaborate?

About secondly: What do you mean by ordinary copying? Using the copy module?